contracts/lazyMint/LazyMintERC1155.sol

```diff
@@ -0,0 +1,606 @@
+    function setPublicMintConditions(
+        uint256 _tokenId, 
+        MintCondition[] calldata _conditions
+    ) 
+        external
+        onlyModuleAdmin 
+    {
+        // make sure the conditions are sorted in ascending order
+        uint256 lastConditionStartTimestamp = 0;
+        uint256 indexForCondition = mintConditions[_tokenId].nextConditionIndex;
+
+        for (uint256 i = 0; i < _conditions.length; i++) {
+            
+            require(
+                lastConditionStartTimestamp == 0 
+                    || lastConditionStartTimestamp < _conditions[i].startTimestamp,
+                "LazyMintERC1155: startTimestamp must be in ascending order."
+            );
+            require(_conditions[i].maxMintSupply > 0, "LazyMintERC1155: max mint supply cannot be 0.");
+            require(_conditions[i].quantityLimitPerTransaction > 0, "LazyMintERC1155: quantity limit cannot be 0.");
+
+            mintConditions[_tokenId].mintConditionAtIndex[indexForCondition] = MintCondition({
+                startTimestamp: _conditions[i].startTimestamp,
+                maxMintSupply: _conditions[i].maxMintSupply,
+                currentMintSupply: 0,
+                quantityLimitPerTransaction: _conditions[i].quantityLimitPerTransaction,
+                waitTimeInSecondsBetweenClaims: _conditions[i].waitTimeInSecondsBetweenClaims,
+                pricePerToken: _conditions[i].pricePerToken,
+                currency: _conditions[i].currency,
+                merkleRoot: _conditions[i].merkleRoot
+            });
+
+            indexForCondition += 1;
+            lastConditionStartTimestamp = _conditions[i].startTimestamp;
+        }
+
+        mintConditions[_tokenId].nextConditionIndex = indexForCondition;
+
+        emit NewMintConditions(_tokenId, _conditions);
+    }
+
+    /// @dev See EIP 2981
+    function royaltyInfo(uint256, uint256 salePrice)
+        external
+        view
+        virtual
+        override
+        returns (address receiver, uint256 royaltyAmount)
+    {
+        receiver = controlCenter.getRoyaltyTreasury(address(this));
+        royaltyAmount = (salePrice * royaltyBps) / MAX_BPS;
+    }
+
+    //      =====   Setter functions  =====
+
+    /// @dev Lets a module admin set the default recipient of all primary sales.
+    function setDefaultSaleRecipient(address _saleRecipient) external onlyModuleAdmin {
+        defaultSaleRecipient = _saleRecipient;
+        emit NewSaleRecipient(_saleRecipient, 0, true);
+    }
+
+    /// @dev Lets a module admin set the recipient of all primary sales for a given token ID.
+    function setDefaultSaleRecipient(uint256 _tokenId, address _saleRecipient) external onlyModuleAdmin {
+        saleRecipient[_tokenId] = _saleRecipient;
+        emit NewSaleRecipient(_saleRecipient, _tokenId, false);
+    }
+
+    /// @dev Lets a module admin pause or unpause the contract.
+    function setPaused(bool _toPause) external onlyModuleAdmin {
```

Comment:
We discussed this issue while writing the `Marketplace` contract. We chose to not have an explicit `PAUSER_ROLE` since it makes sense for the contract to be paused/unpaused directly by a module admin (holder of `DEFAULT_ADMIN_ROLE` on the contract), rather than requiring an additional explicit role.

Reply:
if that's the case, i'm going to remove pause function entirely because it can be achieved using setRestrictedTransfer since it has the same permission scopes.